driver-legacy/src/main/com/mongodb/MongoClient.java

```diff
@@ -142,26 +139,98 @@
      *
      * @return the default codec registry
      * @see MongoClientOptions#getCodecRegistry()
+     * @see MongoClientSettings#getDefaultCodecRegistry()
      * @since 3.0
      */
     public static CodecRegistry getDefaultCodecRegistry() {
         return com.mongodb.MongoClientSettings.getDefaultCodecRegistry();
     }
 
     /**
-     * Creates an instance based on a (single) mongodb node (localhost, default port).
+     * Creates an instance based on a (single) MongoDB server ({@code "mongodb://127.0.0.1:27017"}).
      */
     public MongoClient() {
-        this(new ServerAddress());
+        this(new ConnectionString("mongodb://127.0.0.1"));
+    }
+
+    /**
+     * Creates a MongoClient instance based on a connection string.
+     *
+     * @param connectionString server to connect to in connection string format.  For backwards compatibility, the
```

Comment:
Changed the behavior a bit: this method now supports both a full connection string as well as `host:port`

Reply:
Ack